core/server/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -322,6 +323,9 @@ public void removeBlocks(List<Long> blockIds) {
             }
           }
           mLostBlocks.remove(blockId);
```

Comment:
Good catch. I think you're right:
- for Free, blocks should be added to mLostBlocks
- for delete, blocks should be deleted from both mBlocks and mLostBlocks

I have a follow-up question:
Can one block be shared by multiple files?